frame/contracts/src/exec.rs

```diff
@@ -545,10 +543,18 @@ fn transfer<'a, T: Trait, V: Vm<T>, L: Loader<T>>(
 		Err("not enough gas to pay transfer fee")?
 	}
 
-	// Only ext_terminate is allowed to bring the sender below the existential deposit
+	// Only ext_terminate is allowed to bring the sender below the subsistence
+	// or even existential deposit.
 	let existence_requirement = match cause {
 		Terminate => ExistenceRequirement::AllowDeath,
-		_ => ExistenceRequirement::KeepAlive,
+		_ => {
+			ensure!(
+				T::Currency::total_balance(transactor) >=
```

Comment:
transactor here can actually be an EOA (in case it is the origin) in which case it is actually is not subject for this limitation.

Reply:
Good catch. Totally forgot about that. Was only thinking about contracts calling it though the API.

Reply:
Is fixed now.